lib/bundler/plugin.rb

```diff
@@ -197,7 +197,16 @@ def load_plugin(name)
       # done to avoid conflicts
       path = index.plugin_path(name)
 
+      if insert_index = Bundler.rubygems.load_path_insert_index
```

Comment:
I tried to do that. But could find a way to get `spec` here.

Even if I store the spec's locations, `add_self_to_load_path` depended on `full_gem_path`, which itself depended on `source`.

The alternative to hard coded `lib` is, we can store the `full_required_paths` with index.